### ISSUE TITLE
Native crash when displaying tooltips using Wayland

### ISSUE THREAD

**Issue by rodtell** [flathub/org.eclipse.Java#94]
```
(Eclipse:25): Gdk-CRITICAL **: 09:04:29.822: gdk_window_create_gl_context: assertion 'GDK_IS_WINDOW (window)' failed
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007ff89ed8cb28, pid=25, tid=26
#
# JRE version: OpenJDK Runtime Environment Temurin-17.0.10+7 (17.0.10+7) (build 17.0.10+7)
# Java VM: OpenJDK 64-Bit Server VM Temurin-17.0.10+7 (17.0.10+7, mixed mode, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# Problematic frame:
# C  [libwebkit2gtk-4.1.so.0+0xd8cb28]
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/rodrigo/hs_err_pid25.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/adoptium/adoptium-support/issues
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

```
Just started typing and using snippets completion makes it crash. It became unusable.
Debian 12.5 GNOME environment.

From log generated:
```
OS:
NAME=GNOME
VERSION="45 (Flatpak runtime)"
VERSION_ID=45
ID=org.gnome.Platform
PRETTY_NAME="GNOME 45 (Flatpak runtime)"
BUG_REPORT_URL="https://gitlab.gnome.org/GNOME/gnome-build-meta/-/issues/new"
HOME_URL="https://www.gnome.org/"
uname: Linux 6.1.0-18-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.76-1 (2024-02-01) x86_64
OS uptime: 0 days 2:22 hours
libc: glibc 2.38 NPTL 2.38 
rlimit (soft/hard): STACK 8192k/infinity , CORE 0k/infinity , NPROC 62639/62639 , NOFILE 1048576/1048576 , AS infinity/infinity , CPU infinity/infinity , DATA infinity/infinity , FSIZE infinity/infinity , MEMLOCK 2013248k/2013248k
load average: 1,13 1,69 1,72

/proc/meminfo:
MemTotal:       16105996 kB
MemFree:         1845200 kB
MemAvailable:   10878056 kB
Buffers:          283320 kB
Cached:          9182876 kB
SwapCached:         2352 kB
Active:          2582068 kB
Inactive:       10625584 kB
Active(anon):     611624 kB
Inactive(anon):  3775896 kB
Active(file):    1970444 kB
Inactive(file):  6849688 kB
Unevictable:      124188 kB
Mlocked:            7952 kB
SwapTotal:       1003516 kB
SwapFree:         990716 kB
Zswap:                 0 kB
Zswapped:              0 kB
Dirty:              1668 kB
Writeback:             0 kB
AnonPages:       3667512 kB
Mapped:           596280 kB
Shmem:            640804 kB
KReclaimable:     555272 kB
Slab:             703620 kB
SReclaimable:     555272 kB
SUnreclaim:       148348 kB
KernelStack:       18176 kB
PageTables:        37892 kB
SecPageTables:         0 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     9056512 kB
Committed_AS:    9634680 kB
VmallocTotal:   34359738367 kB
VmallocUsed:       48764 kB
VmallocChunk:          0 kB
Percpu:             6848 kB
HardwareCorrupted:     0 kB
AnonHugePages:    923648 kB
ShmemHugePages:   522240 kB
ShmemPmdMapped:        0 kB
FileHugePages:         0 kB
FilePmdMapped:         0 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
Hugetlb:               0 kB
DirectMap4k:      238764 kB
DirectMap2M:     8927232 kB
DirectMap1G:     8388608 kB

/sys/kernel/mm/transparent_hugepage/enabled: <Not Available>
/sys/kernel/mm/transparent_hugepage/defrag (defrag/compaction efforts parameter): <Not Available>

Process Memory:
Virtual Size: 114674656K (peak: 142587600K)
Resident Set Size: 819964K (peak: 819964K) (anon: 682900K, file: 128140K, shmem: 8924K)
Swapped out: 0K
C-Heap outstanding allocations: 135664K, retained: 109251K
glibc malloc tunables: (default)

/proc/sys/kernel/threads-max (system-wide limit on the number of threads): 125279
/proc/sys/vm/max_map_count (maximum number of memory map areas a process may have): 65530
/proc/sys/kernel/pid_max (system-wide limit on number of process identifiers): 4194304

container information not found.
Steal ticks since vm start: 0
Steal ticks percentage since vm start:  0,000

CPU: total 8 (initial active 8) (4 cores per cpu, 2 threads per core) family 6 model 140 stepping 1 microcode 0xb4, cx8, cmov, fxsr, ht, mmx, 3dnowpref, sse, sse2, sse3, ssse3, sse4.1, sse4.2, popcnt, lzcnt, tsc, tscinvbit, avx, avx2, aes, erms, clmul, bmi1, bmi2, adx, avx512f, avx512dq, avx512cd, avx512bw, avx512vl, sha, fma, vzeroupper, avx512_vpopcntdq, avx512_vpclmulqdq, avx512_vaes, avx512_vnni, clflush, clflushopt, clwb, avx512_vbmi2, avx512_vbmi
CPU Model and flags from /proc/cpuinfo:
model name	: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l2 invpcid_single cdp_l2 ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves split_lock_detect dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq rdpid movdiri movdir64b fsrm avx512_vp2intersect md_clear ibt flush_l1d arch_capabilities

Online cpus: 0-7
Offline cpus: 
BIOS frequency limitation: <Not Available>
Frequency switch latency (ns): 0
Available cpu frequencies: <Not Available>
Current governor: powersave
Core performance/turbo boost: <Not Available>

Memory: 4k page, physical 16105996k(1845200k free), swap 1003516k(990716k free)
Page Sizes: 4k

vm_info: OpenJDK 64-Bit Server VM (17.0.10+7) for linux-amd64 JRE (17.0.10+7), built on Jan 16 2024 00:00:00 by "admin" with gcc 10.3.0

END.

```

**Comment by Mailaender** [comment]
Possibly a duplicate of https://github.com/flathub/org.eclipse.Java/issues/93.

**Comment by rodtell** [comment]
Possible issue related with Wayland. I used [Flatseal](https://flathub.org/apps/com.github.tchx84.Flatseal) and changed the permission to use X11 instead of Wayland, is working without any problems. 
![Captura de tela de 2024-04-04 11-56-05](https://github.com/flathub/org.eclipse.Java/assets/77083219/ee97416e-61f7-46db-ae57-95a626682c4e)

**Comment by rodtell** [comment]
A bit more details on the error, I removed Flatseal (since is a bit overkill for what I need) and used the command:
`flatpak run --nosocket=wayland org.eclipse.Java`

It is definitely a Wayland issue, since blocking Wayland, Eclipse runs smoothly. I also was able to find this error; although again I emphasize that Eclipse is running perfectly with that command first described.
Here is the error that is repeatedly printed on terminal:
```
Gtk-Message: 07:24:26.477: Failed to load module "canberra-gtk-module"
Gtk-Message: 07:24:26.477: Failed to load module "pk-gtk-module"

``` 
It looks like canberra-gtk is a library for UI for pop-ups and other graphical elements (Eclipse when using wayland freezes when opening the Market Place and the tooltips). Hope this is useful, if not, I will delete this comment. Do not want to cause any more confusion  😄

**Comment by Mailaender** [comment]
I also get these

> Failed to load module "canberra-gtk-module"

GTK messages in installations that work fine and also since forever, so I think it is a red herring. No need to delete your comment, though. Just note that these Wayland problems most likely can't be addressed here in the Flatpak packaging. Eclipse crashing with latest WebkitGTK is also an issue for @archlinux and @gentoo, so I think you should report upstream instead.

**Comment by Mailaender** [comment]
https://bugs.eclipse.org/bugs/show_bug.cgi?id=583071

**Comment by rodtell** [comment]
Thank you for the explanation. Next time I will check there first, instead of opening an bug report here. Should I close this issue then since is not related to Flatpak but Eclipse itself?
